packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -149,13 +149,17 @@ export async function deployStack(options: DeployStackOptions): Promise<DeploySt
   const deployName = options.deployName || stackArtifact.stackName;
   let cloudFormationStack = await CloudFormationStack.lookup(cfn, deployName);
 
+  const terminationProtection = stackArtifact.terminationProtection ?? false;
+
   if (!options.force && cloudFormationStack.exists) {
     // bail out if the current template is exactly the same as the one we are about to deploy
     // in cdk-land, this means nothing changed because assets (and therefore nested stacks) are immutable.
-    debug('checking if we can skip this stack based on the currently deployed template and tags (use --force to override)');
+    debug('checking if we can skip this stack based on the currently deployed template, tags and termination protection (use --force to override)');
     const tagsIdentical = compareTags(cloudFormationStack.tags, options.tags ?? []);
-    if (JSON.stringify(stackArtifact.template) === JSON.stringify(await cloudFormationStack.template()) && tagsIdentical) {
-      debug(`${deployName}: no change in template and tags, skipping (use --force to override)`);
+    if (JSON.stringify(stackArtifact.template) === JSON.stringify(await cloudFormationStack.template())
+        && tagsIdentical
+        && cloudFormationStack.terminationProtection === terminationProtection) {
+      debug(`${deployName}: no change in template, tags or termination protection, skipping (use --force to override)`);
```

Comment:
Feels like we should extract all the logic that decides whether a stack should be updated into a separate function now.

Reply:
done